services/job_runner.go

```diff
@@ -214,37 +204,53 @@ func ExecuteRunAtBlock(
 		return jr, wrapError(jr, errors.New("No unfinished tasks to run"))
 	}
 	offset := len(jr.TaskRuns) - len(unfinished)
-	latestRun := unfinished[0]
-
-	merged, err := latestRun.Result.Merge(overrides)
+	prevResult, err := unfinished[0].Result.Merge(jr.Overrides)
 	if err != nil {
 		return jr, wrapError(jr, err)
 	}
-	latestRun.Result = merged
 
 	for i, taskRunTemplate := range unfinished {
-		taskRun, err := taskRunTemplate.MergeTaskParams(overrides.Data)
+		nextTaskRun, err := taskRunTemplate.MergeTaskParams(jr.Overrides.Data)
 		if err != nil {
 			return jr, wrapError(jr, err)
 		}
 
-		latestRun = markCompletedIfRunnable(startTask(jr, taskRun, latestRun.Result, bn, store))
-		jr.TaskRuns[i+offset] = latestRun
-		logTaskResult(latestRun, taskRun, i)
+		lastRun := markCompletedIfRunnable(startTask(jr, nextTaskRun, prevResult, bn, store))
+		jr.TaskRuns[i+offset] = lastRun
+		logTaskResult(lastRun, nextTaskRun, i)
+		prevResult = lastRun.Result
 
 		if err := store.Save(&jr); err != nil {
 			return jr, wrapError(jr, err)
 		}
-		if !latestRun.Status.Runnable() {
+		if !lastRun.Status.Runnable() {
 			break
 		}
 	}
 
-	jr = jr.ApplyResult(latestRun.Result)
+	jr = jr.ApplyResult(prevResult)
 	logger.Infow("Finished current job run execution", jr.ForLogger()...)
 	return jr, wrapError(jr, store.Save(&jr))
 }
 
+func prepareJobRun(jr models.JobRun, store *store.Store, overrides models.RunResult, bn *models.IndexableBlockNumber) (models.JobRun, error) {
+	if jr.Status.CanStart() {
+		jr.Status = models.RunStatusInProgress
+	}
+	var err error
+	jr.Overrides, err = jr.Overrides.Merge(overrides)
+	if err != nil {
+		return jr, err
```

Comment:
Seems like we may want to set the job run status to errored here if we can't merge parameters.

Reply:
I agree. But I also think we should set the job status to errored if anything in this `prepareJobRun` errors and floated the idea of having a `defer` handle all errors inside `ExecuteJobRunAtBlock`. If we're fine with this particular scenario for now I'll add it. But I thought it awkward to cherry pick this error as bubbling up to the job run and none of the others.